starfish/image/_single.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+import collections
+from typing import Tuple
+
+from slicedimage import ImageFormat
+from slicedimage.io import resolve_url
+
+from ._base import ImageBase
+
+
+class SingleImage(ImageBase):
+    def __init__(self, data):
+        self._data = data
+
+    @classmethod
+    def from_url(cls, relativeurl: str, baseurl: str):
+        backend, name, baseurl = resolve_url(relativeurl, baseurl)
+        extension = os.path.splitext(relativeurl)[1].lstrip(".")
+        format = ImageFormat.find_by_extension(extension)
+        with backend.read_file_handle(name) as fh:
+            return SingleImage(format.reader_func(fh))
+
+    @property
+    def numpy_array(self):
+        data = self._data.view()
+        data.setflags(write=False)
```

Comment:
I'd like a comment here about what write=False means for how the class works (since the functionality is in an external repo). 

It makes me think changes to the `SingleImage` won't get written back to the backing file.

Reply:
#197